### ISSUE TITLE
Use Same Bundle for InitFromBundle and SaveStateToBundle

### ISSUE THREAD

**Issue by JimWilcox3** [MvvmCross/MvvmCross#949]
When a ViewModel is possibly being TombStoned, SaveStateToBundle is called.  When the ViewModel is started back up, Init is called before LoadStateFromBundle.  This causes fresh data to be loaded before you even know if you need to restore from a tombstone.  If you used the same bundle for InitFromBundle, then you could check the "parameters" bundle to see if there is saved state data in there and load the saved data from it.  If not, you can use the parameters in that bundle to load fresh data.

Alternatively, you could call LoadStateFromBundle before you call any flavor of Init.  That way you could see if data is present before you load fresh data.

The bottom line is that in the Init, you need the opportunity to load saved data if recovering from a tombstone or load fresh data if it is a new navigation.

**Comment by cyrilcathala** [comment]
You should only load data within the Start() method. Here is what is called (in order) :
- Init() is for initializing navigation parameters.
- LoadStateFromBundle() is for initializing tombstoning parameters.
- Start() should state whether it should load fresh data or not, based on the data retrieved in previous methods

**Comment by vvolkgang** [comment]
@martijn00 about #1165 fixing this, I think @JimWilcox3 needs to change his workflow based on @cyrilcathala comments.

**Comment by JimWilcox3** [comment]
@martijn00 That is what I have done.  I will close this issue.
